src/compiler/compiler.ml

```diff
@@ -304,14 +314,33 @@ let filter ctx tctx =
 	Filters.run ctx.com tctx ctx.com.main;
 	t()
 
+let call_light_init_macro com path =
+	let open MacroContext in
+	let mctx = create_macro_context com in
+	let api = make_macro_com_api com null_pos in
+	let init = create_macro_interp api mctx in
+	MacroContext.MacroLight.call_init_macro com mctx api path;
+	(init,mctx)
+
 let compile ctx actx =
 	let com = ctx.com in
 	(* Set up display configuration *)
 	DisplayProcessing.process_display_configuration ctx;
 	let display_file_dot_path = DisplayProcessing.process_display_file com actx in
+	let mctx = match com.custom_target with
+		| Some (name,_) ->
+			begin try
+				Some (call_light_init_macro com (Printf.sprintf "%s.Init.init()" name))
+			with Error.Error err ->
```

Comment:
Ultimately, we should only catch the case where `target.Init.init()` doesn't exist and silently ignore that. Note that we don't want to ignore all `Error` things because we still want to know about typing errors in the macro.